### PR TITLE
adding experimental change-log for 0.54.0dev0

### PR DESCRIPTION
**Pull request by esc**
Testing using https://github.com/esc/numba-dev-files/blob/master/gitlog2changelog.py to generate a richer change-log with hyperlinks and better attribution.

### CODE REVIEW

**Comment by esc** [comment]
A rendere version is available via ReadTheDocs: https://numba--7154.org.readthedocs.build/en/7154/release-notes.html

**Comment by esc** [comment]
Updated the highlights section and fixed the titles.

**Comment by stuartarchibald** [comment]
From a quick skim of the change log:

Python language support enhancements:
* f-strings (basic support)
* dict comprehension
* `sum` builtin

NumPy features/enhancements
* np.clip
* np.rot90
* np.swapaxes
* np.argmax(axis)
* np.random.dirichlet
* np.iscomplexobj
* np.isreal
* np.isrealobj
* np.isscalar
* np.isneginf
* np.isposinf
* np.iscomplex
* setitem 0d arrays

Debug enhancements:
* DWARF/gdb fixes

A lot of CUDA changes:
* TBC CC @gmarkall 

Experimental target extension work:
* Context managed retargeting of dispatchers.
* Custom NumPy array subclass support.
* Target specific overload support/general mechanism to add new targets.
* Per function compiler flags with inheritance behaviours.
* ``overload_classmethod`` in the extension API.

Version support changes:
- NumPy 1.20 supported
- NumPy minimum is now 1.17
- LLVM 11 on all architectures
- Python minimum is now 3.7
- TBB >= 2021 supported and required to use TBB.
- cloudpickle vendored in for serialisation etc.
- ROCm unmaintained status

File: CHANGE_LOG [hunk 1]
```diff
@@ -1,14 +1,265 @@
...
+  * np.isneginf
+  * np.isposinf
+  * np.iscomplex
+  * setitem 0d arrays
+
+Debug enhancements:
+
+* DWARF/gdb fixes
+* A lot of CUDA changes:
+  * TBC CC @gmarkall
```
Comment (stuartarchibald) [inline CHANGE_LOG:31]: CC @gmarkall to fill this in please !

**Comment by stuartarchibald** [comment]
@esc https://github.com/stuartarchibald/numba/commit/8a8dbd9a4438bc1648fdd75dd95f62aa13a840c9 has some expanded notes if you're happy with them.

**Comment by esc** [comment]
@stuartarchibald thank you, I have merged in the patch and re-generated the PR list.

File: CHANGE_LOG [hunk 2]
```diff
@@ -1,14 +1,291 @@
 Version 0.54.0
```
Comment (stuartarchibald) [inline CHANGE_LOG:1]: Needs the date (see e.g. Version 0.53.0 for format)

Comment (esc) [reply CHANGE_LOG]: done in 386c0fc

**Comment by sklam** [comment]
I'm adding CUDA highlights from @gmarkall at https://github.com/sklam/numba/commit/c21acc01ac7795f21a0e01c05a637a386be03ac3

**Comment by esc** [comment]
@sklam thank you, I have cherrry-picked the patch and applied it. Thank you for the patch!

File: CHANGE_LOG [hunk 3]
```diff
@@ -1,14 +1,318 @@
...
+  * np.iscomplex
+  * np.iscomplexobj
+  * np.isneginf
+  * np.isposinf
+  * np.isreal
+  * np.isrealobj
+  * np.isscalar
+  * np.random.dirichlet
+  * np.rot90
+  * np.swapaxes
```
Comment (stuartarchibald) [inline CHANGE_LOG:26]:
```suggestion
  * ``np.clip``
  * ``np.iscomplex``
  * ``np.iscomplexobj``
  * ``np.isneginf``
  * ``np.isposinf``
  * ``np.isreal``
  * ``np.isrealobj``
  * ``np.isscalar``
  * ``np.random.dirichlet``
  * ``np.rot90``
  * ``np.swapaxes``
```
Should probably put `` around these?

File: CHANGE_LOG [hunk 4]
```diff
@@ -1,14 +1,318 @@
...
+
+Also ``np.argmax`` has gain support for the ``axis`` keyword argument and it's
```
Comment (stuartarchibald) [inline CHANGE_LOG:28]:
```suggestion
Also ``np.argmax`` has gained support for the ``axis`` keyword argument and it's
```

File: CHANGE_LOG [hunk 5]
```diff
@@ -1,14 +1,318 @@
...
+Internal changes:
+
+* Debugging support through DWARF has been fixed and enhanced.
+* Numba now optimises the way in which locals are emitted to help reduce time
+  spend in LLVM's SROA passes.
+
+CUDA target changes:
 
-In development
+* Support for emitting lineinfo to be consumed by profiling tools such as
```
Comment (stuartarchibald) [inline CHANGE_LOG:39]:
```suggestion
* Support for emitting ``lineinfo`` to be consumed by profiling tools such as
```

File: CHANGE_LOG [hunk 6]
```diff
@@ -1,14 +1,318 @@
...
+  * When making implicit copies by calling a kernel on arrays in host memory
+  * When occupancy is poor due to kernel or ufunc/gufunc configuration
 
-* Minimum python version raised to 3.7.
+* Support for implementing warp-aggregated intrinsics:
+
+  * Using support for more CUDA functions: ``activemask()``, ``lanemask_lt()``
+  * The ``ffs()`` function now works correctly!
+
+* Support for @overload in the CUDA target
```
Comment (stuartarchibald) [inline CHANGE_LOG:56]:
```suggestion
* Support for ``@overload`` in the CUDA target
```

File: CHANGE_LOG [hunk 7]
```diff
@@ -1,14 +1,318 @@
...
+
+Intel kindly sponsored research and development that lead to a number of new
+features and internal support changes:
+
+* Dispatchers can now be retargetted to a new target via a use defined context
```
Comment (stuartarchibald) [inline CHANGE_LOG:61]:
```suggestion
* Dispatchers can now be retargetted to a new target via a user defined context
```

File: CHANGE_LOG [hunk 8]
```diff
@@ -1,14 +1,318 @@
...
+* Per function compiler flags with inheritance behaviours.
+* The extension API now has support for overloading class methods via the
+  ``@overload_classmethod`` decorator.
+
+Deprecations:
+
+* The ``ROCm`` target (for AMD ROC GPUs) has been moved to an "unmaintained"
+  status and a seperate repository stub has been created for it at:
+  https://github.com/numba/numba-rocm
+* Multiple CUDA related deprecations TBC
```
Comment (stuartarchibald) [inline CHANGE_LOG:76]:
```suggestion
```

File: CHANGE_LOG [hunk 9]
```diff
@@ -1,14 +1,318 @@
...
+
+CUDA target deprecations and breaking changes:
+
+* Relaxed strides checking is now the default when computing contiguity of
```
Comment (stuartarchibald) [inline CHANGE_LOG:80]:
```suggestion
* Relaxed strides checking is now the default when computing the contiguity of
```

File: CHANGE_LOG [hunk 10]
```diff
@@ -1,14 +1,318 @@
...
+  device arrays.
+* The ``inspect_ptx()`` method is deprecated. For use cases that obtain PTX for
+  further compilation outside of Numba, use ``compile_ptx()`` instead.
+* Eager compilation of device functions (the case when ``device=True`` and a
+  signature is provided) is deprecated.
+
+Version support/dependency changes:
+
+* LLVM 11 now supported on all platforms via llvmlite.
```
Comment (stuartarchibald) [inline CHANGE_LOG:89]:
```suggestion
* LLVM 11 is now supported on all platforms via llvmlite.
```

File: CHANGE_LOG [hunk 11]
```diff
@@ -1,14 +1,318 @@
...
+* The minimum supported Python version is raised to 3.7.
 * NumPy version 1.20 supported.
```
Comment (stuartarchibald) [inline CHANGE_LOG:91]:
```suggestion
* NumPy version 1.20 is supported.
```

File: CHANGE_LOG [hunk 12]
```diff
@@ -1,14 +1,318 @@
...
-* Minimum supported NumPy version raised to 1.17.
-* Moved ROCm target to "unmaintained" status.
+* The minimum supported NumPy version raised to 1.17 for runtime (compilation
```
Comment (stuartarchibald) [inline CHANGE_LOG:92]:
```suggestion
* The minimum supported NumPy version is raised to 1.17 for runtime (compilation
```

File: CHANGE_LOG [hunk 13]
```diff
@@ -1,14 +1,318 @@
...
+  however remains compatible with NumPy 1.11).
+* Vendor `cloudpickle <https://github.com/cloudpipe/cloudpickle>`_ `v1.6.0` --
+  now used for all `pickle` operations.
```
Comment (stuartarchibald) [inline CHANGE_LOG:95]:
```suggestion
  now used for all ``pickle`` operations.
```

File: CHANGE_LOG [hunk 14]
```diff
@@ -1,14 +1,318 @@
...
+* PR `#6468 <https://github.com/numba/numba/pull/6468>`_: Access `replace_functions_map` via PreParforPass instance (`Sergey Pokhodenko <https://github.com/PokhodenkoSA>`_ `Reazul Hoque <https://github.com/reazulhoque>`_)
+* PR `#6469 <https://github.com/numba/numba/pull/6469>`_: Add address space in pointer type (`Sergey Pokhodenko <https://github.com/PokhodenkoSA>`_ `Reazul Hoque <https://github.com/reazulhoque>`_)
+* PR `#6608 <https://github.com/numba/numba/pull/6608>`_: Support f-strings for common cases (`Ehsan Totoni <https://github.com/ehsantn>`_)
+* PR `#6619 <https://github.com/numba/numba/pull/6619>`_: Improved fastmath code generation for trig, log, and exp/pow. (`Graham Markall <https://github.com/gmarkall>`_ `Michael Collison <https://github.com/testhound>`_)
+* PR `#6680 <https://github.com/numba/numba/pull/6680>`_: Allow getattr to work in objmode output type spec (`Siu Kwan Lam <https://github.com/sklam>`_)
+* PR `#6681 <https://github.com/numba/numba/pull/6681>`_: Explicitly catch `with..as` and raise error. (`stuartarchibald <https://github.com/stuartarchibald>`_)
```
Comment (stuartarchibald) [inline CHANGE_LOG:109]:
```suggestion
* PR `#6681 <https://github.com/numba/numba/pull/6681>`_: Explicitly catch ``with..as`` and raise error. (`stuartarchibald <https://github.com/stuartarchibald>`_)
```

File: CHANGE_LOG [hunk 15]
```diff
@@ -1,14 +1,318 @@
...
+  now used for all `pickle` operations.
+* TBB >= 2021 is now supported and all prior versions are unsupported (not
+  easily possible to maintain the ABI breaking changes).
+
+Pull-Requests:
+
+* PR `#4516 <https://github.com/numba/numba/pull/4516>`_: Make setitem accept 0d np-arrays (`Guilherme Leobas <https://github.com/guilhermeleobas>`_)
+* PR `#4610 <https://github.com/numba/numba/pull/4610>`_: Implement np.is* functions (`Guilherme Leobas <https://github.com/guilhermeleobas>`_)
+* PR `#5984 <https://github.com/numba/numba/pull/5984>`_: Handle idx and size unification in wrap_index manually. (`Todd A. Anderson <https://github.com/DrTodd13>`_)
+* PR `#6468 <https://github.com/numba/numba/pull/6468>`_: Access `replace_functions_map` via PreParforPass instance (`Sergey Pokhodenko <https://github.com/PokhodenkoSA>`_ `Reazul Hoque <https://github.com/reazulhoque>`_)
```
Comment (stuartarchibald) [inline CHANGE_LOG:104]:
```suggestion
* PR `#6468 <https://github.com/numba/numba/pull/6468>`_: Access ``replace_functions_map`` via PreParforPass instance (`Sergey Pokhodenko <https://github.com/PokhodenkoSA>`_ `Reazul Hoque <https://github.com/reazulhoque>`_)
```

File: CHANGE_LOG [hunk 16]
```diff
@@ -1,14 +1,318 @@
...
+* PR `#6755 <https://github.com/numba/numba/pull/6755>`_: install llvmlite from numba/label/dev (`esc <https://github.com/esc>`_)
+* PR `#6758 <https://github.com/numba/numba/pull/6758>`_: patch to compile _devicearray.cpp with c++11 (`esc <https://github.com/esc>`_)
+* PR `#6759 <https://github.com/numba/numba/pull/6759>`_: Update CHANGE_LOG for 0.53.0rc2 (`stuartarchibald <https://github.com/stuartarchibald>`_)
+* PR `#6760 <https://github.com/numba/numba/pull/6760>`_: Fix scheduler bug where it rounds to 0 divisions for a chunk. (`Todd A. Anderson <https://github.com/DrTodd13>`_)
+* PR `#6762 <https://github.com/numba/numba/pull/6762>`_: Glue wrappers to create @overload from split typing and lowering. (`stuartarchibald <https://github.com/stuartarchibald>`_ `Siu Kwan Lam <https://github.com/sklam>`_)
+* PR `#6766 <https://github.com/numba/numba/pull/6766>`_: Fix DeviceNDArray null shape issue (`Michael Collison <https://github.com/testhound>`_)
+* PR `#6769 <https://github.com/numba/numba/pull/6769>`_: CUDA: Replace `CachedPTX` and `CachedCUFunction` with `CUDACodeLibrary` functionality (`Graham Markall <https://github.com/gmarkall>`_)
```
Comment (stuartarchibald) [inline CHANGE_LOG:137]:
```suggestion
* PR `#6769 <https://github.com/numba/numba/pull/6769>`_: CUDA: Replace ``CachedPTX`` and ``CachedCUFunction`` with ``CUDACodeLibrary`` functionality (`Graham Markall <https://github.com/gmarkall>`_)
```

File: CHANGE_LOG [hunk 17]
```diff
@@ -1,14 +1,318 @@
...
+* PR `#6798 <https://github.com/numba/numba/pull/6798>`_: CUDA: Add optional Driver API argument logging (`Graham Markall <https://github.com/gmarkall>`_)
+* PR `#6799 <https://github.com/numba/numba/pull/6799>`_: Print Numba and llvmlite versions in sysinfo (`Graham Markall <https://github.com/gmarkall>`_)
+* PR `#6800 <https://github.com/numba/numba/pull/6800>`_: Make a common standard API for querying ufunc impl (`Sergey Pokhodenko <https://github.com/PokhodenkoSA>`_ `Siu Kwan Lam <https://github.com/sklam>`_)
+* PR `#6801 <https://github.com/numba/numba/pull/6801>`_: ParallelAccelerator no long will convert StaticSetItem to SetItem because record arrays require StaticSetItems. (`Todd A. Anderson <https://github.com/DrTodd13>`_)
+* PR `#6802 <https://github.com/numba/numba/pull/6802>`_: Add lineinfo flag to PTX and SASS compilation (`Graham Markall <https://github.com/gmarkall>`_ `Max Katz <https://github.com/maxpkatz>`_)
+* PR `#6804 <https://github.com/numba/numba/pull/6804>`_: added runtime version to `numba -s` (`Kalyan <https://github.com/rawwar>`_)
```
Comment (stuartarchibald) [inline CHANGE_LOG:151]:
```suggestion
* PR `#6804 <https://github.com/numba/numba/pull/6804>`_: added runtime version to ``numba -s`` (`Kalyan <https://github.com/rawwar>`_)
```

File: CHANGE_LOG [hunk 18]
```diff
@@ -1,14 +1,318 @@
...
+* PR `#6808 <https://github.com/numba/numba/pull/6808>`_: #3468 continued: Add support for `np.clip` (`Graham Markall <https://github.com/gmarkall>`_ `Aaron Russell Voelker <https://github.com/arvoelke>`_)
```
Comment (stuartarchibald) [inline CHANGE_LOG:152]:
```suggestion
* PR `#6808 <https://github.com/numba/numba/pull/6808>`_: #3468 continued: Add support for ``np.clip`` (`Graham Markall <https://github.com/gmarkall>`_ `Aaron Russell Voelker <https://github.com/arvoelke>`_)
```

File: CHANGE_LOG [hunk 19]
```diff
@@ -1,14 +1,318 @@
...
+* PR `#6911 <https://github.com/numba/numba/pull/6911>`_: CUDA: Add support for activemask(), lanemask_lt(), and nanosleep() (`Graham Markall <https://github.com/gmarkall>`_)
+* PR `#6912 <https://github.com/numba/numba/pull/6912>`_: Prevent use of varargs in closure calls. (`stuartarchibald <https://github.com/stuartarchibald>`_)
+* PR `#6913 <https://github.com/numba/numba/pull/6913>`_: Add runtests option to gitdiff on the common ancestor (`Siu Kwan Lam <https://github.com/sklam>`_)
+* PR `#6915 <https://github.com/numba/numba/pull/6915>`_: Update _Intrinsic for sphinx to capture the inner docstring (`Guilherme Leobas <https://github.com/guilhermeleobas>`_)
+* PR `#6917 <https://github.com/numba/numba/pull/6917>`_: Add type conversion for StringLiteral to unicode_type and test. (`stuartarchibald <https://github.com/stuartarchibald>`_)
+* PR `#6918 <https://github.com/numba/numba/pull/6918>`_: Start section on commonly encounted unsupported parfors code. (`stuartarchibald <https://github.com/stuartarchibald>`_)
+* PR `#6924 <https://github.com/numba/numba/pull/6924>`_: CUDA: Fix `ffs` (`Graham Markall <https://github.com/gmarkall>`_)
```
Comment (stuartarchibald) [inline CHANGE_LOG:202]:
```suggestion
* PR `#6924 <https://github.com/numba/numba/pull/6924>`_: CUDA: Fix ``ffs`` (`Graham Markall <https://github.com/gmarkall>`_)
```

File: CHANGE_LOG [hunk 20]
```diff
@@ -1,14 +1,318 @@
...
+* PR `#6929 <https://github.com/numba/numba/pull/6929>`_: Fix CI failure when gitpython is missing. (`Siu Kwan Lam <https://github.com/sklam>`_)
+* PR `#6935 <https://github.com/numba/numba/pull/6935>`_: fixes broken link in numba-runtime.rst (`Kalyan <https://github.com/rawwar>`_)
+* PR `#6936 <https://github.com/numba/numba/pull/6936>`_: CUDA: Implement support for PTDS globally (`Graham Markall <https://github.com/gmarkall>`_)
+* PR `#6937 <https://github.com/numba/numba/pull/6937>`_: Fix memory leak in bytes boxing (`stuartarchibald <https://github.com/stuartarchibald>`_)
+* PR `#6940 <https://github.com/numba/numba/pull/6940>`_: Fix function resolution for intrinsics across hardware. (`stuartarchibald <https://github.com/stuartarchibald>`_)
+* PR `#6941 <https://github.com/numba/numba/pull/6941>`_: ABC the target descriptor and make consistent throughout. (`stuartarchibald <https://github.com/stuartarchibald>`_)
+* PR `#6944 <https://github.com/numba/numba/pull/6944>`_: CUDA: Support for `@overload` (`Graham Markall <https://github.com/gmarkall>`_)
```
Comment (stuartarchibald) [inline CHANGE_LOG:210]:
```suggestion
* PR `#6944 <https://github.com/numba/numba/pull/6944>`_: CUDA: Support for ``@overload`` (`Graham Markall <https://github.com/gmarkall>`_)
```

File: CHANGE_LOG [hunk 21]
```diff
@@ -1,14 +1,318 @@
...
+* PR `#6945 <https://github.com/numba/numba/pull/6945>`_: Fix issue with array analysis tests needing scipy. (`stuartarchibald <https://github.com/stuartarchibald>`_)
+* PR `#6948 <https://github.com/numba/numba/pull/6948>`_: Refactor registry init. (`stuartarchibald <https://github.com/stuartarchibald>`_ `Graham Markall <https://github.com/gmarkall>`_ `Siu Kwan Lam <https://github.com/sklam>`_)
+* PR `#6953 <https://github.com/numba/numba/pull/6953>`_: CUDA: Fix and deprecate `inspect_ptx()`, fix NVVM option setup for device functions (`Graham Markall <https://github.com/gmarkall>`_)
```
Comment (stuartarchibald) [inline CHANGE_LOG:213]:
```suggestion
* PR `#6953 <https://github.com/numba/numba/pull/6953>`_: CUDA: Fix and deprecate ``inspect_ptx()``, fix NVVM option setup for device functions (`Graham Markall <https://github.com/gmarkall>`_)
```

File: CHANGE_LOG [hunk 22]
```diff
@@ -1,14 +1,318 @@
...
+* PR `#6958 <https://github.com/numba/numba/pull/6958>`_: Inconsistent behavior of reshape between numpy and numba/cuda device array (`Lauren Arnett <https://github.com/laurenarnett>`_)
+* PR `#6961 <https://github.com/numba/numba/pull/6961>`_: Update overload glue to deal with typing_key (`stuartarchibald <https://github.com/stuartarchibald>`_)
+* PR `#6964 <https://github.com/numba/numba/pull/6964>`_: Move minimum supported Python version to 3.7 (`stuartarchibald <https://github.com/stuartarchibald>`_)
+* PR `#6966 <https://github.com/numba/numba/pull/6966>`_: Fix issue with TBB test detecting forks from incorrect state. (`stuartarchibald <https://github.com/stuartarchibald>`_)
+* PR `#6971 <https://github.com/numba/numba/pull/6971>`_: Fix CUDA @``intrinsic`` use (`stuartarchibald <https://github.com/stuartarchibald>`_)
```
Comment (stuartarchibald) [inline CHANGE_LOG:218]:
```suggestion
* PR `#6971 <https://github.com/numba/numba/pull/6971>`_: Fix CUDA ``@intrinsic`` use (`stuartarchibald <https://github.com/stuartarchibald>`_)
```

File: CHANGE_LOG [hunk 23]
```diff
@@ -1,14 +1,318 @@
...
+* PR `#7056 <https://github.com/numba/numba/pull/7056>`_: Change prange conversion approach to reuse header block. (`Todd A. Anderson <https://github.com/DrTodd13>`_)
+* PR `#7061 <https://github.com/numba/numba/pull/7061>`_: Add ndarray allocator classmethod (`stuartarchibald <https://github.com/stuartarchibald>`_ `Siu Kwan Lam <https://github.com/sklam>`_)
+* PR `#7064 <https://github.com/numba/numba/pull/7064>`_: Testhound/host array performance warning (`Michael Collison <https://github.com/testhound>`_)
+* PR `#7066 <https://github.com/numba/numba/pull/7066>`_: Fix #7065: Add expected exception messages for NumPy 1.20 to tests (`Graham Markall <https://github.com/gmarkall>`_)
+* PR `#7068 <https://github.com/numba/numba/pull/7068>`_: Enhancing docs about PRNG seeding (`Jérome Eertmans <https://github.com/jeertmans>`_)
+* PR `#7070 <https://github.com/numba/numba/pull/7070>`_: Improve the issue templates and pull request template. (`Guoqiang QI <https://github.com/guoqiangqi>`_)
+* PR `#7080 <https://github.com/numba/numba/pull/7080>`_: Fix `__eq__` for Flags and cpu_options classes (`Siu Kwan Lam <https://github.com/sklam>`_)
```
Comment (stuartarchibald) [inline CHANGE_LOG:254]:
```suggestion
* PR `#7080 <https://github.com/numba/numba/pull/7080>`_: Fix ``__eq__`` for Flags and cpu_options classes (`Siu Kwan Lam <https://github.com/sklam>`_)
```

File: CHANGE_LOG [hunk 24]
```diff
@@ -1,14 +1,318 @@
...
+* PR `#7122 <https://github.com/numba/numba/pull/7122>`_: Rtd/sphinx compat (`esc <https://github.com/esc>`_)
+* PR `#7134 <https://github.com/numba/numba/pull/7134>`_: Move minimum LLVM version to 11. (`stuartarchibald <https://github.com/stuartarchibald>`_)
+* PR `#7137 <https://github.com/numba/numba/pull/7137>`_: skip pycc test on Python 3.7 + macOS because of distutils issue (`esc <https://github.com/esc>`_)
+* PR `#7138 <https://github.com/numba/numba/pull/7138>`_: Update the Azure default linux image to Ubuntu 18.04 (`stuartarchibald <https://github.com/stuartarchibald>`_)
+* PR `#7141 <https://github.com/numba/numba/pull/7141>`_: Require llvmlite 0.37 as minimum supported. (`stuartarchibald <https://github.com/stuartarchibald>`_)
+* PR `#7143 <https://github.com/numba/numba/pull/7143>`_: Update version checks in __init__ for np 1.17 (`stuartarchibald <https://github.com/stuartarchibald>`_)
+* PR `#7145 <https://github.com/numba/numba/pull/7145>`_: Fix mainline (`stuartarchibald <https://github.com/stuartarchibald>`_)
+* PR `#7146 <https://github.com/numba/numba/pull/7146>`_: Fix `inline_closurecall` may not be imported (`Siu Kwan Lam <https://github.com/sklam>`_)
```
Comment (stuartarchibald) [inline CHANGE_LOG:270]:
```suggestion
* PR `#7146 <https://github.com/numba/numba/pull/7146>`_: Fix ``inline_closurecall`` may not be imported (`Siu Kwan Lam <https://github.com/sklam>`_)
```

File: CHANGE_LOG [hunk 25]
```diff
@@ -1,14 +1,318 @@
...
+* PR `#7147 <https://github.com/numba/numba/pull/7147>`_: Revert "Workaround gitpython 3.1.18 dependency issue" (`stuartarchibald <https://github.com/stuartarchibald>`_)
+* PR `#7149 <https://github.com/numba/numba/pull/7149>`_: Fix issue in bytecode analysis where target and next are same. (`stuartarchibald <https://github.com/stuartarchibald>`_)
+* PR `#7152 <https://github.com/numba/numba/pull/7152>`_: Fix iterators in CUDA (`Graham Markall <https://github.com/gmarkall>`_)
+* PR `#7156 <https://github.com/numba/numba/pull/7156>`_: Fix `ir_utils._max_label` being updated incorrectly (`Siu Kwan Lam <https://github.com/sklam>`_)
```
Comment (stuartarchibald) [inline CHANGE_LOG:274]:
```suggestion
* PR `#7156 <https://github.com/numba/numba/pull/7156>`_: Fix ``ir_utils._max_label`` being updated incorrectly (`Siu Kwan Lam <https://github.com/sklam>`_)
```

File: CHANGE_LOG [hunk 26]
```diff
@@ -1,14 +1,318 @@
...
+* PR `#6961 <https://github.com/numba/numba/pull/6961>`_: Update overload glue to deal with typing_key (`stuartarchibald <https://github.com/stuartarchibald>`_)
+* PR `#6964 <https://github.com/numba/numba/pull/6964>`_: Move minimum supported Python version to 3.7 (`stuartarchibald <https://github.com/stuartarchibald>`_)
+* PR `#6966 <https://github.com/numba/numba/pull/6966>`_: Fix issue with TBB test detecting forks from incorrect state. (`stuartarchibald <https://github.com/stuartarchibald>`_)
+* PR `#6971 <https://github.com/numba/numba/pull/6971>`_: Fix CUDA @``intrinsic`` use (`stuartarchibald <https://github.com/stuartarchibald>`_)
+* PR `#6977 <https://github.com/numba/numba/pull/6977>`_: Vendor cloudpickle (`Siu Kwan Lam <https://github.com/sklam>`_)
+* PR `#6978 <https://github.com/numba/numba/pull/6978>`_: Implement operator.contains for empty Tuples (`Brandon T. Willard <https://github.com/brandonwillard>`_)
+* PR `#6981 <https://github.com/numba/numba/pull/6981>`_: Fix LLVM IR parsing error on use of `np.bool_` in globals (`stuartarchibald <https://github.com/stuartarchibald>`_)
```
Comment (stuartarchibald) [inline CHANGE_LOG:221]:
```suggestion
* PR `#6981 <https://github.com/numba/numba/pull/6981>`_: Fix LLVM IR parsing error on use of ``np.bool_`` in globals (`stuartarchibald <https://github.com/stuartarchibald>`_)
```

**Review by stuartarchibald — COMMENTED** [review]
@esc I've just scanned the rendered doc for markup issues. Suggest the following along with a few typo fixes.

File: CHANGE_LOG [hunk 27]
```diff
@@ -1,14 +1,318 @@
...
+* PR `#6841 <https://github.com/numba/numba/pull/6841>`_: CUDA: Fix source location on kernel entry and enable breakpoints to be set on kernels by mangled name (`Graham Markall <https://github.com/gmarkall>`_)
+* PR `#6843 <https://github.com/numba/numba/pull/6843>`_: cross-referenced Array type in docs (`Kalyan <https://github.com/rawwar>`_)
+* PR `#6844 <https://github.com/numba/numba/pull/6844>`_: CUDA: Remove NUMBAPRO env var warnings, envvars.py + other small tidy-ups (`Graham Markall <https://github.com/gmarkall>`_)
+* PR `#6848 <https://github.com/numba/numba/pull/6848>`_: Ignore .ycm_extra_conf.py (`Graham Markall <https://github.com/gmarkall>`_)
+* PR `#6849 <https://github.com/numba/numba/pull/6849>`_: Add __hash__ for IntEnum (`Hannes Pahl <https://github.com/HPLegion>`_)
+* PR `#6850 <https://github.com/numba/numba/pull/6850>`_: Fix up more internal warnings (`stuartarchibald <https://github.com/stuartarchibald>`_)
+* PR `#6851 <https://github.com/numba/numba/pull/6851>`_: set non-reported llvm timing values to 0.0 (`Ben Greiner <https://github.com/bnavigator>`_)
```
Comment (stuartarchibald) [inline CHANGE_LOG:176]:
```suggestion
```
was in 0.53.1

File: CHANGE_LOG [hunk 28]
```diff
@@ -1,14 +1,318 @@
...
+* PR `#6813 <https://github.com/numba/numba/pull/6813>`_: Mostly CUDA: Replace llvmpy API usage with llvmlite APIs (`Graham Markall <https://github.com/gmarkall>`_)
+* PR `#6814 <https://github.com/numba/numba/pull/6814>`_: Improving context stack (`stuartarchibald <https://github.com/stuartarchibald>`_ `Siu Kwan Lam <https://github.com/sklam>`_)
+* PR `#6817 <https://github.com/numba/numba/pull/6817>`_: Update changelog for 0.53.0 final release date (`Siu Kwan Lam <https://github.com/sklam>`_)
+* PR `#6818 <https://github.com/numba/numba/pull/6818>`_: CUDA: Support IPC on Windows (`Graham Markall <https://github.com/gmarkall>`_)
+* PR `#6822 <https://github.com/numba/numba/pull/6822>`_: Add support for np.rot90 (`stuartarchibald <https://github.com/stuartarchibald>`_ `Daniel Nagel <https://github.com/braniii>`_)
+* PR `#6826 <https://github.com/numba/numba/pull/6826>`_: Fix regression on gufunc serialization (`Siu Kwan Lam <https://github.com/sklam>`_)
+* PR `#6828 <https://github.com/numba/numba/pull/6828>`_: Fix regression in CUDA: Set stream in mapped and managed array device_setup (`Graham Markall <https://github.com/gmarkall>`_)
```
Comment (stuartarchibald) [inline CHANGE_LOG:162]:
```suggestion
```
was in 0.53.1

File: CHANGE_LOG [hunk 29]
```diff
@@ -1,14 +1,318 @@
...
+* PR `#6829 <https://github.com/numba/numba/pull/6829>`_: Fix accuracy of np.arange and np.linspace (`stuartarchibald <https://github.com/stuartarchibald>`_)
+* PR `#6830 <https://github.com/numba/numba/pull/6830>`_: CUDA: Use relaxed strides checking to compute contiguity (`Graham Markall <https://github.com/gmarkall>`_)
+* PR `#6833 <https://github.com/numba/numba/pull/6833>`_: Raise TypeError exception if numpy array is cast to scalar (`Michael Collison <https://github.com/testhound>`_)
+* PR `#6834 <https://github.com/numba/numba/pull/6834>`_: Remove illegal "debug" kw argument (`Shaun Cutts <https://github.com/shaunc>`_)
+* PR `#6836 <https://github.com/numba/numba/pull/6836>`_: CUDA: Documentation updates (`Graham Markall <https://github.com/gmarkall>`_)
+* PR `#6837 <https://github.com/numba/numba/pull/6837>`_: Ignore warnings from packaging module when testing import behaviour. (`stuartarchibald <https://github.com/stuartarchibald>`_)
```
Comment (stuartarchibald) [inline CHANGE_LOG:168]:
```suggestion
```
was in 0.53.1

File: CHANGE_LOG [hunk 30]
```diff
@@ -1,14 +1,318 @@
...
+* PR `#6811 <https://github.com/numba/numba/pull/6811>`_: CUDA: Remove test of runtime being a supported version (`Graham Markall <https://github.com/gmarkall>`_)
+* PR `#6813 <https://github.com/numba/numba/pull/6813>`_: Mostly CUDA: Replace llvmpy API usage with llvmlite APIs (`Graham Markall <https://github.com/gmarkall>`_)
+* PR `#6814 <https://github.com/numba/numba/pull/6814>`_: Improving context stack (`stuartarchibald <https://github.com/stuartarchibald>`_ `Siu Kwan Lam <https://github.com/sklam>`_)
+* PR `#6817 <https://github.com/numba/numba/pull/6817>`_: Update changelog for 0.53.0 final release date (`Siu Kwan Lam <https://github.com/sklam>`_)
+* PR `#6818 <https://github.com/numba/numba/pull/6818>`_: CUDA: Support IPC on Windows (`Graham Markall <https://github.com/gmarkall>`_)
+* PR `#6822 <https://github.com/numba/numba/pull/6822>`_: Add support for np.rot90 (`stuartarchibald <https://github.com/stuartarchibald>`_ `Daniel Nagel <https://github.com/braniii>`_)
+* PR `#6826 <https://github.com/numba/numba/pull/6826>`_: Fix regression on gufunc serialization (`Siu Kwan Lam <https://github.com/sklam>`_)
```
Comment (stuartarchibald) [inline CHANGE_LOG:161]:
```suggestion
```
was in 0.53.1

**Review by stuartarchibald — COMMENTED** [review]
Suggested duplicate PR removal. This are in the released 0.53.1.

File: CHANGE_LOG [hunk 31]
```diff
@@ -1,14 +1,313 @@
...
---------------
+Version 0.54.0 (30 June, 2021)
+------------------------------
+
+This release includes a significant number of new features, important
+refactoring, critical bug fixes and a number of dependency upgrades.
+
+Python language support enhancements:
+
+* Basic support for ``f-strings``,
```
Comment (stuartarchibald) [inline CHANGE_LOG:9]:
```suggestion
* Basic support for ``f-strings``.
```

File: CHANGE_LOG [hunk 32]
```diff
@@ -1,14 +1,313 @@
...
+* ``dict`` comprehensions are now supported.
+* The ``sum`` built-in function is implemented.
+
+NumPy features/enhancements:
+
+The following functions are now supported.
```
Comment (stuartarchibald) [inline CHANGE_LOG:15]:
```suggestion
The following functions are now supported:

```
else it's in bold font

File: docs/environment.yml [hunk 33]
```diff
@@ -10,5 +10,7 @@ dependencies:
   - numpy
   - numpydoc
   - setuptools
+  # https://stackoverflow.com/questions/67542699/readthedocs-sphinx-not-rendering-bullet-list-from-rst-fileA
+  - docutils==0.16
```
Comment (stuartarchibald) [inline docs/environment.yml:14]: Is this now effectively a dependency?

Comment (stuartarchibald) [reply docs/environment.yml:14]: No. It's a workaround for a bug.

**Review by sklam — APPROVED** [review]
LGTM

**Pull request merged**